.github/workflows/cli-verification.yaml

```diff
@@ -0,0 +1,57 @@
+name: CLI Publish & Verify
+
+on:
+    push:
+        branches:
+            - master
+            - staging/**
+    pull_request:
+
+concurrency:
+    group: verify-cli-${{ github.event.pull_request.number || github.ref }}
+    cancel-in-progress: true
+
+jobs:
+    publish-and-test:
+        runs-on: ubuntu-latest
+        env:
+            NANGO_CLI_UPGRADE_MODE: ignore
+        permissions:
+            contents: read
+            packages: write
+        steps:
+            - uses: actions/checkout@v4
+
+            - uses: actions/setup-node@v4
+              with:
+                  node-version: '18.x'
+                  registry-url: 'https://npm.pkg.github.com'
+                  scope: '@nangohq'
+                  always-auth: true
+            - name: Publish npm packages to the github registry
+              env:
+                  NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              shell: bash
+              continue-on-error: true
+              run: |
+                  GIT_HASH=$(git rev-parse HEAD)
+                  bash ./scripts/publish.sh 0.0.1-$GIT_HASH
+            - name: Publish the cli privately under the correct scope
+              working-directory: packages/cli
+              env:
+                  NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              run: |
+                  jq '.name = "@nangohq/cli"' package.json > temp.json && mv temp.json package.json
+                  npm publish --access public
+            - name: Install the cli from the github package registry
+              env:
+                  NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              run: |
+                  GIT_HASH=$(git rev-parse HEAD)
+                  mkdir nango-cli-test && cd nango-cli-test
+                  npm init -y
+                  npm install @nangohq/cli@0.0.1-$GIT_HASH
+                  npx nango version --debug
```

Comment:
I would at least test the output

Reply:
Don't see much gain there

Reply:
if you don't test how can you be sure it's the right version being tested? also not all errors will throw process.exit(1)